### PR TITLE
Set up GitHub Actions for testing

### PR DESCRIPTION
**Pull request by DilumAluthge**

### CODE REVIEW

**Comment by DilumAluthge** [comment]
@StefanKarpinski Looks like you don't have a `docs/make.jl` file?

**Comment by DilumAluthge** [comment]
Anyway, CI is passing. Just the docs failing.

Btw, you should add a (Base64-encoded) private key as a GitHub secret. Click on the Settings tab of this repo, click on "Secrets" in the left hand sidebar. And then add the (Base64-encoded) private key as a secret named `DOCUMENTER_KEY`. And then make sure to add the corresponding public key as a deploy key with write access on this repo.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      fail-fast: false
+      matrix:
+        version:
+          - '1.3'
+          - '1' # automatically expands to the latest stable 1.x release of Julia.
+          - 'nightly'
+        os:
+          - ubuntu-latest
+        arch:
+          - x64
```
Comment (giordano) [inline .github/workflows/ci.yml:24]: I guess it's better to test on as many platforms as possible?
```suggestion
        os:
          - ubuntu-latest
          - macos-latest
          - windows-latest
        julia-arch:
          - x64
          - x86
        # 32-bit Julia binaries are not available on macOS
        exclude:
          - os: macOS-latest
            julia-arch: x86
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      matrix:
+        version:
+          - '1.3'
+          - '1' # automatically expands to the latest stable 1.x release of Julia.
+          - 'nightly'
+        os:
+          - ubuntu-latest
+          - macos-latest
+          - windows-latest
+        julia-arch:
```
Comment (giordano) [inline .github/workflows/ci.yml:25]: My bad:
```suggestion
        arch:
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+          - ubuntu-latest
+          - macos-latest
+          - windows-latest
+        arch:
+          - x64
+          - x86
+        # 32-bit Julia binaries are not available on macOS
+        exclude:
+          - os: macOS-latest
+            julia-arch: x86
```
Comment (giordano) [inline .github/workflows/ci.yml:31]: My bad 2:
```suggestion
            arch: x86
```

**Comment by DilumAluthge** [comment]
I've added an empty `docs/make.jl` file just so CI doesn't complain.

We can populate that file with the appropriate contents later.

**Comment by DilumAluthge** [comment]
@StefanKarpinski This should be good to go once all CI jobs are green.

**Comment by DilumAluthge** [comment]
🎉 All green now.

**Comment by StefanKarpinski** [comment]
Why do I need a `docs/make.jl` file?

**Comment by DilumAluthge** [comment]
I just assumed you had docs for this package, because I saw that the `/docs/src` directory existed.

If you don't have docs, we can delete the entire `/docs` directory, and I'll comment out the docs step from the GitHub Actions config.

**Comment by DilumAluthge** [comment]
I've removed the Documentation job.

**Comment by StefanKarpinski** [comment]
Right. The reason for the `docs/src/index.md` stub is because Julia's `make docs` looks there when generating stdlib docs for this. As a standalone package the docs are just the README.

**Comment by StefanKarpinski** [comment]
Is the UUID change script here so that it can be tested on nightly? If so, 👏, both clever and thorough!

**Comment by DilumAluthge** [comment]
> Right. The reason for the `docs/src/index.md` stub is because Julia's `make docs` looks there when generating stdlib docs for this. As a standalone package the docs are just the README.

Ahh. I had assumed that this package was similar to Pkg.jl, where we build the docs as part of CI.

**Comment by StefanKarpinski** [comment]
Nah, for packages with such simple APIs, I just put all the docs in the README.

**Pull request merged**